Makefile

```diff
@@ -774,5 +773,15 @@ $(GO_APIDIFF_BIN): $(GO_APIDIFF)
 $(YQ): ## Build yq from tools folder.
 	GOBIN=$(TOOLS_BIN_DIR) $(GO_INSTALL) github.com/mikefarah/yq/v4 $(YQ_BIN) $(YQ_VER)
 
+$(SETUP_ENVTEST): # Build setup-envtest from tools folder.
+	GOBIN=$(TOOLS_BIN_DIR) $(GO_INSTALL) $(SETUP_ENVTEST_PKG) $(SETUP_ENVTEST_BIN) $(SETUP_ENVTEST_VER)
+	@if [ $(shell go env GOOS) == "darwin" ]; then \
+		$(eval KUBEBUILDER_ASSETS := $(shell $(SETUP_ENVTEST) use --use-env -p path --arch amd64 $(KUBEBUILDER_ENVTEST_KUBERNETES_VERSION))) \
```

Comment:
is there a arm64 version available now? I see [here](https://github.com/kubernetes-sigs/cluster-api/blob/main/Makefile#L596) that CAPI has a comment to use amd64 until arm64 is available

Reply:
Like you say, I was working off `master` from cluster-api as a reference.  I'll adjust this for now so that the build passes and if there's further adjustments or suggestions I can iterate.

Reply:
My apologies–I just now stumbled across this comment from weeks ago.

`make go-test` on this branch fails on my Mac M1:

```shell
% uname -snrm && make clean go-test
Darwin katie.local 21.5.0 arm64
...
GOBIN=/Users/matt/projects/cluster-api-provider-azure/hack/tools/bin ./scripts/go_install.sh sigs.k8s.io/controller-runtime/tools/setup-envtest setup-envtest v0.0.0-20211110210527-619e6b92dab9
rm: /Users/matt/projects/cluster-api-provider-azure/hack/tools/bin/setup-envtest*: No such file or directory
unable to fetch checksum for requested version: unable fetch metadata for kubebuilder-tools-1.20.2-darwin-arm64.tar.gz -- got status "404 Not Found" from GCS
KUBEBUILDER_ASSETS="" go test ./... 
?   	sigs.k8s.io/cluster-api-provider-azure	[no test files]
ok  	sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3	16.939s
ok  	sigs.k8s.io/cluster-api-provider-azure/api/v1alpha4	18.974s
ok  	sigs.k8s.io/cluster-api-provider-azure/api/v1beta1	1.846s
...
Running Suite: Controller Suite
===============================
Random Seed: 1656528736
Will run 2 of 2 specs

STEP: bootstrapping test environment
Panic [0.022 seconds]
[BeforeSuite] BeforeSuite 
/Users/matt/projects/cluster-api-provider-azure/controllers/suite_test.go:47

  Test Panicked
  unable to start control plane itself: failed to start the controlplane. retried 5 times: exec: "etcd": executable file not found in $PATH
  /Users/matt/projects/cluster-api-provider-azure/internal/test/env/env.go:105

  Full Stack Trace
  sigs.k8s.io/cluster-api-provider-azure/internal/test/env.NewTestEnvironment()
  	/Users/matt/projects/cluster-api-provider-azure/internal/test/env/env.go:105 +0x300
  sigs.k8s.io/cluster-api-provider-azure/controllers.glob..func3(0x140001386c0)
  	/Users/matt/projects/cluster-api-provider-azure/controllers/suite_test.go:49 +0x40
  reflect.Value.call({0x105f94fe0, 0x1063073d8, 0x13}, {0x105b8201a, 0x4}, {0x1400009af50, 0x1, 0x1})
  	/usr/local/go/src/reflect/value.go:556 +0x5a8
  reflect.Value.Call({0x105f94fe0, 0x1063073d8, 0x13}, {0x14000215f50, 0x1, 0x1})
  	/usr/local/go/src/reflect/value.go:339 +0x8c
  github.com/onsi/ginkgo/internal/leafnodes.newRunner.func1(0x140001386c0)
  	/Users/matt/go/pkg/mod/github.com/onsi/ginkgo@v1.16.5/internal/leafnodes/runner.go:49 +0x1bc
  github.com/onsi/ginkgo/internal/leafnodes.(*runner).runAsync.func1(0x1400040d320, 0x140001386c0)
  	/Users/matt/go/pkg/mod/github.com/onsi/ginkgo@v1.16.5/internal/leafnodes/runner.go:86 +0x80
  created by github.com/onsi/ginkgo/internal/leafnodes.(*runner).runAsync
  	/Users/matt/go/pkg/mod/github.com/onsi/ginkgo@v1.16.5/internal/leafnodes/runner.go:71 +0x7c
------------------------------


Ran 2 of 0 Specs in 0.022 seconds
FAIL! -- 0 Passed | 2 Failed | 0 Pending | 0 Skipped
--- FAIL: TestAPIs (0.02s)
FAIL
FAIL	sigs.k8s.io/cluster-api-provider-azure/controllers	0.500s
?   	sigs.k8s.io/cluster-api-provider-azure/exp	[no test files]
ok  	sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3	12.861s
ok  	sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha4	14.457s
ok  	sigs.k8s.io/cluster-api-provider-azure/exp/api/v1beta1	2.769s
Running Suite: Controller Suite
===============================
Random Seed: 1656528736
Will run 1 of 1 specs

STEP: bootstrapping test environment
Panic [0.008 seconds]
[BeforeSuite] BeforeSuite 
/Users/matt/projects/cluster-api-provider-azure/exp/controllers/suite_test.go:50

  Test Panicked
  unable to start control plane itself: failed to start the controlplane. retried 5 times: exec: "etcd": executable file not found in $PATH
  /Users/matt/projects/cluster-api-provider-azure/internal/test/env/env.go:105

  Full Stack Trace
  sigs.k8s.io/cluster-api-provider-azure/internal/test/env.NewTestEnvironment()
  	/Users/matt/projects/cluster-api-provider-azure/internal/test/env/env.go:105 +0x300
  sigs.k8s.io/cluster-api-provider-azure/exp/controllers.glob..func2(0x1400017da40)
  	/Users/matt/projects/cluster-api-provider-azure/exp/controllers/suite_test.go:52 +0x50
  reflect.Value.call({0x101d02fa0, 0x1020eeb48, 0x13}, {0x1018cb728, 0x4}, {0x1400020ef50, 0x1, 0x1})
  	/usr/local/go/src/reflect/value.go:556 +0x5a8
  reflect.Value.Call({0x101d02fa0, 0x1020eeb48, 0x13}, {0x14000084750, 0x1, 0x1})
  	/usr/local/go/src/reflect/value.go:339 +0x8c
  github.com/onsi/ginkgo/internal/leafnodes.newRunner.func1(0x1400017da40)
  	/Users/matt/go/pkg/mod/github.com/onsi/ginkgo@v1.16.5/internal/leafnodes/runner.go:49 +0x1bc
  github.com/onsi/ginkgo/internal/leafnodes.(*runner).runAsync.func1(0x140008aed80, 0x1400017da40)
  	/Users/matt/go/pkg/mod/github.com/onsi/ginkgo@v1.16.5/internal/leafnodes/runner.go:86 +0x80
  created by github.com/onsi/ginkgo/internal/leafnodes.(*runner).runAsync
  	/Users/matt/go/pkg/mod/github.com/onsi/ginkgo@v1.16.5/internal/leafnodes/runner.go:71 +0x7c
...
% file ./hack/tools/bin/setup-envtest   
./hack/tools/bin/setup-envtest: Mach-O 64-bit executable arm64
```